src/Admin/AbstractAdmin.php

```diff
@@ -2724,16 +2728,49 @@ public function isGranted($name, $object = null)
 
     public function getUrlSafeIdentifier($model)
     {
+        // NEXT_MAJOR: Remove this check and use object as param typehint.
+        if (null === $model) {
+            @trigger_error(sprintf(
+                'Passing null as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will be not allowed in version 4.0.',
+                __METHOD__
+            ), E_USER_DEPRECATED);
+
+            return null;
+        }
+
         return $this->getModelManager()->getUrlSafeIdentifier($model);
     }
 
     public function getNormalizedIdentifier($model)
```

Comment:
There is an open question: https://github.com/sonata-project/SonataAdminBundle/pull/6311#issuecomment-678785483

Should I remove these deprecations because the interface does not allow null.
Or should I update the interface because `null` was sometimes use as argument.

Reply:
This method is inherit from `AdminInterface` where `@param mixed $model` and `@return string`. Technicly `null` is not allowed as param or return type hint. I know `null` will passing becouse docblock can be ignore. IMHO we should keep `trigger_error` and remove `return null;` line. This fix PHPStan, display deprecation for people and allow use `ModelManager` with null $model argument. Of course we should add `trigger_error` in model to then.

Reply:
I thought against about it, and we don't have to add deprecation for someone which does not respect the phpdoc, or we will start to add deprecation for every methods.